lib/renaissance/helpers/constraint.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Renaissance.Helpers.Constraint do
+  import Ecto.Changeset
+  alias Renaissance.Helpers.Adapt
+  alias Renaissance.Auctions
+
+  def amount_constraint(changeset, field) do
+    auction_id = get_change(changeset, :auction_id, 0)
+
+    proposed = get_change(changeset, field, 0) |> Adapt.money_value()
+    current = Auctions.get_current_amount(auction_id) |> Adapt.money_value()
+
+    if proposed > current do
+      changeset
+    else
+      message = ~s(must be greater than #{Money.new(current)})
+      add_error(changeset, field, message)
+    end
+  end
+
+  def bidder_constraint(changeset, field) do
+    bidder_id = get_change(changeset, field)
+    auction_id = get_change(changeset, :auction_id, 0)
+    seller_id = Auctions.get_seller_id(auction_id)
+
+    if is_nil(seller_id) or bidder_id != seller_id do
```

Comment:
This condition is a little bit hard to parse. I guess I would expect a little bit more clarity on the condition that we actually care about here:

```
case {seller_id, bidder_id} do
  {nil, _} -> changeset # How do we even end up in this place?
  {seller_id, bidder_id} when seller_id == bidder_id -> add_error(changeset, "can't bid on the item you're selling")
  _ -> changeset
end
```
end
```